File: python/rotateArray.py

def reverseArray(arr, low, high):
    while (low < high):
        temp = arr[low]
        arr[low] = arr[high]
        arr[high] = temp
        low = low + 1
        high = high - 1

def rotateArray(arr, k):
    low = 0
    high = len(arr)
    reverseArray(arr, low, high - 1)
    reverseArray(arr, low, k -1)
    reverseArray(arr, k, high - 1)
    return arr

arr = [1,2,3,4,5,6,7]
out = rotateArray(arr, 1)
print(out)

File: Java/RotateArray.java
public class RotateArray {
    public static void main(String[] args) {
        int arr[] = { 6, 3, 0, 1, 0, 4, 4, 7, 8, 0 };
        int k = 2;
        int low = 0, high = arr.length - 1;
        reverse(arr, low, high);
        reverse(arr, low, k - 1);
        reverse(arr, k, high);

        for (int i = 0; i < arr.length; i++) {
            System.out.println(arr[i]);
        }
    }

    public static void reverse(int[] arr, int low, int high) {
        while (low < high) {
            int temp = arr[low];
            arr[low] = arr[high];
            arr[high] = temp;
            low++;
            high--;
        }
    }
}
